tests/CountryTest.php

```diff
@@ -9,43 +9,22 @@ class CountryTest extends TestCase
 {
     /**
      * @test
+     * @dataProvider provideInvalidData
      */
-    public function invalidConstructionThrowsException(): void
+    public function invalidConstructionThrowsException(array $data, array $expected): void
     {
-        $isoAlpha2 = 'ZZ';
-        $isoAlpha3 = 'XYZ';
-        $isoNumeric = 9999;
-        $commonName = 'Common Name';
-        $officialName = 'Official Name';
-        $continent = 'Asia';
-        $topLevelDomain = 'tld';
-        $longDistancePrefix = 99;
-        $currencyIsoAlpha = 'XYD';
-
-        $this->expectException(InvalidArgumentException::class);
+        $this->expectException($expected[0]);
+        $this->expectExceptionMessage($expected[1]);
```

Comment:
`expectExceptionObject`?

Reply:
Well, it's matter of habit. I'll change.